internal/errors/runtime_test.go

```diff
@@ -0,0 +1,416 @@
+//
+// Copyright (C) 2019-2021 vdaas.org vald team <vald@vdaas.org>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package errors
+
+import (
+	"fmt"
+	"math"
+	"runtime"
+	"testing"
+	"time"
+
+	"go.uber.org/goleak"
+)
+
+func TestErrPanicRecovered(t *testing.T) {
+	t.Parallel()
+	type args struct {
+		err error
+		rec interface{}
+	}
+	type want struct {
+		want error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return Errorf("got: \"%#v\",\n\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	defaultErr := New("runtime error")
+	tests := []test{
+		func() test {
+			r := math.MaxFloat64
+			return test{
+				name: "return an error when err is not empty and rec is int value",
+				args: args{
+					err: defaultErr,
+					rec: r,
+				},
+				want: want{
+					want: Wrap(defaultErr, Errorf("panic recovered: %v", r).Error()),
```

Comment:
Should we use New function insted of Wrap function.
because this want is equal to the test target function like this.
https://github.com/vdaas/vald/blob/1b6993be91fc53f20a143bc650e546686f97394c/internal/errors/runtime.go#L25

Reply:
if we'd like to test the `Wrap` function, I agree with you.
IMO, about this case, it is not needed.
How do you think?

Reply:
I got it 👍 I agree with you!